tests/TestCase/Http/ResponseTest.php

```diff
@@ -741,6 +741,21 @@ public function testCheckNotModifiedNoHints()
         $this->assertFalse($response->checkNotModified($request));
     }
 
+    /**
+     * Test setting the cookie collection
+     *
+     * @return void
+     */
+    public function testWithCookieCollection()
+    {
+        $response = new Response();
+
+        $collection = new CookieCollection();
+        $newResponse = $response->withCookieCollection($collection);
+
+        $this->assertNotSame($response->getCookieCollection(), $newResponse->getCookieCollection());
```

Comment:
You might also want to assert that the responses are not the same instance, and that the old response didn't get mutated.